### PR TITLE
Fix acceptance tests and restore installation of .h files

### PR DESCRIPTION
**Pull request by erykoff**
The previous PR #98 actually shouldn't have passed tests because it was calling `acceptArray1` which was doing a comparison vs a known vector, rather than just testing acceptance.  This PR fixes the test (which mysteriously was passing in some instances even though it shouldn't).

This also restores the installation of the .h files which is (at least) needed by conda-forge and had been removed in https://github.com/ndarray/ndarray/commit/d151f9fadf75775d58bbbbb18a1450eec5c0cf35

### CODE REVIEW

File: tests/pybind11_test_mod.cc [hunk 1]
```diff
@@ -41,6 +41,8 @@ bool acceptArray1(ndarray::Array<double,1,1> const & a1) {
 
 void acceptArray10(ndarray::Array<double,1,0> const & a1) {}
 
+void acceptArray11(ndarray::Array<double,1,1> const & a1) {}
```
Comment (TallJimbo) [inline tests/pybind11_test_mod.cc:44]: I have no one to blame but myself, but this all started to go wrong when that first `acceptArrayXY` snuck in with no meaningful implementation, as when I reviewed the last PR I certainly wasn't expecting this to be what was in the body of it.  And thinking about it more I better how this passes the test without really getting ndarray to the point of supporting zero shape elements.

I should probably try to find the time to make the behavior of those arrays just more consistent generally, but I really don't have time now.  In the meantime, could you maybe just rename `acceptArray10` and `acceptArray11` to something like `acceptZeroShapeArrayXY` to make it clear that they're more different from the other `acceptArray` methods?

Comment (erykoff) [reply tests/pybind11_test_mod.cc:44]: The naming of these is certainly confusing.  I thought that `accept` meant "pybind11 will allow this as an override" but some of the other ones are actually checking values.  But I'll rename right now

**Comment by erykoff** [comment]
@TallJimbo I think this is ready?
